src/predict.jl

```diff
@@ -33,7 +31,6 @@ function predict(
     estimator::LGBMEstimator, X::AbstractMatrix{TX}; predict_type::Integer = 0,
     start_iteration::Integer = 0, num_iterations::Integer = -1, verbosity::Integer = 1,
     is_row_major::Bool = false,
-    num_threads::Integer = -1,
```

Comment:
I removed a separate num_threads parameter from `predict()` because it is a valid model parameter which can be taken directly from the model (either the default or what user decided to set up) without having to pass additionally pass it.  This way we could end up with quite a few params that could be added to predict() for e.g. the pred early stop or the predict type params but all of them can be read from the model.

Reply:
brough back `num_threads` in predictions as this allows to specify number of threads specifically for predictions which is separate to the number of threads used in training (which can be provided through `estimator.num_threads`